src/ocr.py

```diff
@@ -0,0 +1,290 @@
+from fileinput import close
+from tesserocr import PyTessBaseAPI, PSM, OEM
+import numpy as np
+import cv2
+import re
+import csv
+import difflib
+from utils.misc import erode_to_black
+
+from logger import Logger
+from typing import List, Union
+from dataclasses import dataclass
+
+@dataclass
+class OcrResult:
+    text: str = None
+    original_text: str = None
+    word_confidences: list = None
+    mean_confidence: float = None
+    # these are kept to help train OCR
+    original_img: np.ndarray = None
+    processed_img: np.ndarray = None
+    def __getitem__(self, key):
+        return super().__getattribute__(key)
+
+class Ocr:
```

Comment:
I'd try to add some encapsulation to this class by prefixing methods which are not intended to be called by others with "_"